code/modules/clothing/suits/marine_armor.dm

```diff
@@ -184,7 +184,7 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 		new_look = GLOB.armor_style_list[user.client.prefs.preferred_armor]
 
 	icon_state = replacetext(icon_state,"1","[new_look]")
-	update_icon()
+	update_icon(user ? user : null)
```

Comment:
```suggestion
	update_icon(user)
```
Unless theres some other nuance to this, such as 0 being a possibility and it must be null, its unnecessary to perform this check if all you're going to return is null.

Reply:
It doesn't update the onmob for the armor for ridged armor for some godforsaken reason unless I do this.